webpack/assets/javascripts/foreman_filters.js

```diff
@@ -0,0 +1,25 @@
+import $ from 'jquery';
+
+export function resourceTypeChanged(type) {
+  $.ajax({
+    url: type.dataset.url,
+    data: {
+      resource_type: type.value,
+    },
+    dataType: 'script',
+  });
+}
+
+export function unlimitedChanged(checkbox) {
+  const isChecked = $(checkbox).prop('checked');
+  $('#search').prop('disabled', isChecked);
+}
+
+export function overrideTaxonomyChecked(checkbox) {
+  const isChecked = $(checkbox).prop('checked');
+  const data = $('#filter_resource_type').data();
+  const { allowOrganizations, allowLocations } = data;
```

Comment:
when the resource type changes, it updates the `data-allow-locations` attr (https://github.com/theforeman/foreman/blob/develop/app/views/permissions/index.js.erb#L27), so allowLocations doesn't get updated. the lines there need to be changes to `.data('allow-location', <%=...` for this to work or otherwise changing between resources will not refresh the tabs that should be displayed when changing this checkbox.

Reply:
@tbrisker updated, but I wasn't sure how to test it?

Reply:
edit a filter on a resource that only allows one, e.g. locations, change the resource type to one that allows both, eg domain, and see that the clicking the override button shows just the location tab, then change to one that doesn't have any, such as hw model and click the override button again and see that it still shows locations

Reply:
I think you forgot to add the change to the commit?